fragalysis_api/xcimporter/align.py

```diff
@@ -98,138 +111,407 @@ def __get_length_and_resolution(self, file):
-        pymol_cmd = self._load_objs()
-
-        for num, name in enumerate(pymol_cmd.get_names()):  # Saves the aligned pdb files from the cmd as pdb files
-
-            if not name == self._get_ref:
-                pymol_cmd.align(name, self._get_ref)
-                self._save_align(name, pymol_cmd.get_pdbstr(selection=name), out_dir)
-
-            elif name == self._get_ref:
-                self._save_align(name, pymol_cmd.get_pdbstr(selection=name), out_dir)
+        # load ref
+        input_files = self._get_files
+        map_list = self._get_maplist
+        crystals = [os.path.splitext(os.path.basename(f))[0] for f in input_files]
+        ref = self._get_ref
+        dir = self.directory
+        mono = self.mono
+
+        s = time.time()
+        for num, name in enumerate(crystals):
+            reference_pdb = Structure.from_file(file=Path(os.path.join(dir, f'{ref}.pdb')))
+            all_maps = [j for j in map_list if name in j]
+            if not name == ref:
+                # Do an alignment + save
+                current_pdb = Structure.from_file(file=Path(os.path.join(dir, f'{name}.pdb')))
+                try:
+                    current_pdb, transform = current_pdb.align_to(other=reference_pdb, monomerized=mono)
+                except Exception as e:
+                    # Poorly Documented. Use better stuff...
+                    print(f'{e}')
+                    continue
+
+                current_pdb.structure.write_pdb(os.path.join(out_dir, f'{name}_bound.pdb'))
+                # Align Xmaps + save!
+                for i in all_maps:
+                    base, ext = os.path.splitext(os.path.basename(i))
+                    print(i)
+                    # self.read_reshape_resave(name=base, out_dir=out_dir, ext=ext, transform=transform)
+                    s2 = time.time()
+                    map = Xmap.from_file(file=Path(os.path.join(dir, f'{base}{ext}')))
+                    map.resample(xmap=map, transform=transform)
+                    map.save(path=Path(os.path.join(out_dir, f'{base}{ext}')))
+                    e2 = time.time()
+                    print(f'{int(e2 - s2)} seconds to transform map...')
+
+                    e = time.time()
+                    print(f'Total Running time: {int(e - s) / 60} minutes.')
+
+            else:
+                shutil.copyfile(os.path.join(self.directory, f'{name}.pdb'),
+                                os.path.join(out_dir, f'{name}_bound.pdb'))
+                if os.path.exists(os.path.join(self.directory, f'{name}_smiles.txt')):
+                    shutil.copyfile(os.path.join(self.directory, f'{name}_smiles.txt'),
+                                    os.path.join(out_dir, f'{name}_smiles.txt'))
+                for i in all_maps:
+                    base, ext = os.path.splitext(os.path.basename(i))
+                    shutil.copyfile(i, os.path.join(out_dir, f"{base}{ext}"))
+
+
+# Conor's stuff
+@dataclasses.dataclass()
```

Comment:
Is this stuff not included in the ``pandda_gemmi`` repo? If it isn't - we should get @ConorFWild to put it there - it's probably useful to a wider audience than us anyway. If it is, can you not just import it?

Reply:
It is included but I had to modify one of the class methods.  I will create a branch on pandda_gemmi and shove it in. No idea how much it will affect what Conor's stuff does though.